test/extensions/filters/http/wasm/test_data/http_callout_cpp.cc

```diff
@@ -0,0 +1,50 @@
+// NOLINT(namespace-envoy)
+#include <string>
+#include <unordered_map>
+
+#include "proxy_wasm_intrinsics.h"
+
+class ExampleContext : public Context {
+ public:
+  explicit ExampleContext(uint32_t id, RootContext *root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders() override;
+};
+static RegisterContextFactory
+    register_ExampleContext(CONTEXT_FACTORY(ExampleContext));
+
+FilterHeadersStatus ExampleContext::onRequestHeaders() {
+  try {
+    WasmDataPtr data = getRequestHeader("x-callout-url");
+
+    if (!data) {
+      logWarn("Missing x-callout-url header, cannot forward");
+      return FilterHeadersStatus::Continue;
+    }
+
+    StringView callout_url{data->toString()};
+
+    logInfo("Forwarding to: " + std::string(callout_url));
```

Comment:
@jplevyak take note.  After 20 or so iterations the callout_url will log something that looks corrupted

Reply:
It looks like this is disable for the exception issue.